pkg/tsdb/prometheus/prometheus.go

```diff
@@ -73,6 +73,18 @@ func newInstanceSettings(httpClientProvider httpclient.Provider, cfg *setting.Cf
 	}
 }
 
+func getOauthTokenForCallResource(headersIn map[string][]string) map[string]string {
+	headers := make(map[string]string)
+	accessValues := headersIn["Authorization"]
```

Comment:
This should be attached only when Forward Oauth attached. Or am I missing something?

Reply:
Yes, it happens in the provider.
https://github.com/grafana/grafana/blob/main/pkg/tsdb/prometheus/client/provider.go#L65

So we just pass in the headers regardless... The provider will then strip them out according to whether the option is enabled.